pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -359,18 +359,22 @@ func syncNamespace(
 	kubeClient clientset.Interface,
 	clientPool dynamic.ClientPool,
 	opCache operationNotSupportedCache,
-	groupVersionResources []unversioned.GroupVersionResource,
+	groupVersionResourcesFn func() ([]unversioned.GroupVersionResource, error),
 	namespace *api.Namespace,
 	finalizerToken api.FinalizerName,
 ) error {
+	groupVersionResources, err := groupVersionResourcesFn()
```

Comment:
this has the potential to be called many, many times (syncNamespace requeues the namespace in error/conflict cases)... at least move it just above the call to `deleteAllContent` so we don't fetch these unnecessarily. Also, I thought there were multiple discovery calls to look up the available resources, not just a single API call.